### PR TITLE
React: Expose button to delete analysis on the front-end

### LINKED ISSUE #1051: Expose delete button for analyses to admins
We already have the `DELETE /api/analyses/:id` endpoint for admins only, but it's not exposed in the frontend. Add another button on the analysis page or modal visible to admins for this action. It should give an appropriate warning like the dataset delete button.

### PR DESCRIPTION
**Pull request by hannahle**
Closes #1051

### CODE REVIEW

File: react/src/Analyses/index.tsx [hunk 1]
```diff
@@ -621,6 +669,14 @@ export default function Analyses() {
...
                             },
                         },
+                        {
+                            icon: Delete,
+                            tooltip: "Delete Analysis",
+                            onClick: (event, rowData) => {
+                                setActiveRows(rowData as Analysis[]);
+                                setConfirmDelete(true);
+                            },
+                        },
```
Comment (kevinlul) [inline react/src/Analyses/index.tsx:679]: Can you hide this if the user is not an admin?

**Review by kevinlul — CHANGES_REQUESTED** [review]

File: react/src/Analyses/index.tsx (hunk 1 above)
Comment (hannahle) [reply react/src/Analyses/index.tsx:679]: On it!

Comment (kevinlul) [reply react/src/Analyses/index.tsx]: Sorry, I meant to physically hide the button, which is done on the datasets page with its delete button.

**Review by kevinlul — APPROVED** [review]
LGTM 🚀

**Pull request merged**